dataset/map_view.cpp

```diff
@@ -80,18 +117,19 @@ template <class Key, class Value> void Dict<Key, Value>::rebuildSizes() {
   m_sizes = std::move(new_sizes);
 }
 
+// TODO remove force workaround required by DataArray::slice
```

Comment:
@jl-wynen Can you think of a nice way for avoiding this flag? I have considered adding a `setReadonly(bool)`, or private helper methods and add `DataArray` and `Dataset` as friends, but I don't like either approach.

Reply:
You could add a function that merges two `Dict`s and in https://github.com/scipp/scipp/blob/587475c203c570bb8e148ade3ae723e9923291ea/dataset/data_array.cpp#L94 use
```.cpp
Attrs unaligned_coords;
for (auto it = m_coords.begin(); it != m_coords.end(); ++it)
  if (unaligned_by_dim_slice(*it, s))
    unaligned_coords.set(it->first, out.m_coords.extract(it->first));
return DataArray{m_data->slice(s), m_coords.slice(s), m_masks->slice(s),
                 merge(m_attrs->slice(s), std::move(unaligned_attrs)), m_name};
```

Reply:
Good idea, I went with something like that (got only slightly messy for datasets), see last commit I pushed.